app/api/files/exportRoutes.ts

```diff
@@ -0,0 +1,86 @@
+import { createWriteStream, unlink } from 'fs';
+import errorLog from 'api/log/errorLog';
+import { search } from 'api/search';
+import { validation } from '../utils';
+import { isArray } from 'util';
+import { temporalFilesPath, generateFileName } from './filesystem';
+import { CSVExporter } from 'api/csv';
+import { User } from 'api/users/usersModel';
+import settings from 'api/settings';
+
+type Middleware = (
+  req: { user: User; language: any; query: any },
+  res: { download: any },
+  next: any
+) => Promise<void> | void;
+type Get = { (arg0: string, arg1: Middleware, arg2: Middleware): void };
+type App = { get: Get };
```

Comment:
I duplicated the strategy used in another route because I needed the request to have those extra attributes, but I'm sure there's a better way using Express types.
I'll try to improve it.